Iterable/iterable-web-sdk
# [Part 0] In-App Cypress Tests - In-App UI

## JIRA Ticket(s) if any

* [MOB-3928](https://iterable.atlassian.net/browse/MOB-3928)

## Description

Adds in-app UI to React app for requesting in-app messages and displaying them automatically

## Screenshots

<img width="1178" alt="Screen Shot 2022-05-02 at 11 05 58 AM" src="https://user-images.githubusercontent.com/7387001/166300948-d76bcb83-82de-433f-8c02-7a8f0ef529aa.png">

## Test Steps

1. Run a local JWT generation backend by [cloning this repo](https://github.com/Iterable/jwt-generator) and running `docker-compose up -d`
2. Run React example app in this repo with `yarn install:all && yarn start:all:react`
3. Open up `localhost:8080` in your browser and login in the navbar
4. Click on "inApp" button to go to inapp controls
5. Click on "Get Messages (do not auto-display)" and ensure request is made and output is displayed on the right
6. Click on "Get Messages (auto-display)" and ensure the messages show automatically
    * also ensure pause button prevents new messages from showing
    * and resume button shows the next message after pausing
